R/save.R

```diff
@@ -0,0 +1,66 @@
+#' Save TensorFlow Model
+#' 
+#' Saves a TensorFlow model with the goal of deploying and reusing across systems.
+#' 
+#' @param model A TensorFlow model. For instance, as generated by \code{linear_regressor()}.
+#' @param path The destination path where the model will be saved.
+#' @param spec The model specification, each model has a default specification;
+#'   but it can be further defined specific functions like \code{tf_save_regressor_spec}.
+#'   
+#' @family save
+#' @export
+tf_save <- function(model, path, spec) {
+  UseMethod("tf_save")
+}
+
+#' Create Save-Specification from Model
+#' 
+#' Creates default specification for a given model.
+#' 
+#' @param model A TensorFlow model. For instance, as generated by \code{linear_regressor()}.
+#' 
+#' @export
+#' @family save
+tf_save_spec_from_model <- function(model) {
+  UseMethod("tf_save_spec_from_model")
+}
+
+#' @export
+#' @family save
+tf_save_spec_from_model.tf_estimator <- function(model) {
+  feature_columns <- c(
+    model$params$feature_columns,
+    model$params$linear_feature_columns,
+    model$params$dnn_feature_columns
```

Comment:
I am not sure why you are extracting all those three types of columns but then this function only needs one of those three - `model$params$feature_columns`. Also other estimators may not have `linear_feature_column` or `dnn_feature_column` so maybe move construct this at specific canned estimator level?